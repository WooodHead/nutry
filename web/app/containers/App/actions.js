import {
  SEARCH_COMPLETE,
} from 'containers/Search/constants';

import {
  DEFAULT_ACTION,
  LOGIN_CLICKED,
  LOGIN_SUCCESS,
  LOGIN_FAILURE,
} from './constants';


export function defaultAction() {
  return {
    type: DEFAULT_ACTION,
  };
}
export function login() {
  return {
    type: LOGIN_CLICKED,
  };
}
export function loginSuccess(token) {
  return {
    type: LOGIN_SUCCESS,
    token,
  };
}

export function searchComplete(searchResults) {
  return {
    type: SEARCH_COMPLETE,
    searchResults,
  };
}

export function loginFailure(err) {
  return {
    type: LOGIN_FAILURE,
    err,
  };
}
