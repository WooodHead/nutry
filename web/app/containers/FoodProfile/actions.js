/*
 *
 * FoodProfile actions
 *
 */
import { defaultPortions } from 'lib/nutrientMap';
import {
  DEFAULT_ACTION,
  ON_BACK,
  GET_PROFILE,
  TAB_CHANGED,
  PORTION_CHANGED,
  AGE_GROUP_CHANGED,
  NUTRIENT_SELECTED,
  GET_PROFILE_FAILURE,
  GET_PROFILE_SUCCESS,
} from './constants';


export function defaultAction() {
  return {
    type: DEFAULT_ACTION,
  };
}
export function loadProfile(serialNumber, source) {
  return {
    type: GET_PROFILE,
    serialNumber,
    source,
  };
}
export function loadProfileSuccess(profileBody) {
  return {
    type: GET_PROFILE_SUCCESS,
    nutrientsById: profileBody.nutrients,
    portionsAvailable: defaultPortions(profileBody.portions),
  };
}
export function loadProfileFailure(error) {
  return {
    type: GET_PROFILE_FAILURE,
    error,
  };
}
export function tabChanged(tab) {
  return {
    type: TAB_CHANGED,
    tab,
  };
}
export function nutrientSelected(prefix, id) {
  return {
    type: NUTRIENT_SELECTED,
    prefix,
    id,
  };
}
export function portionChanged(portionSelected) {
  return {
    type: PORTION_CHANGED,
    portionSelected,
  };
}
export function ageGroupChanged(ageGroupSelected) {
  return {
    type: AGE_GROUP_CHANGED,
    ageGroupSelected,
  };
}

export function onBack() {
  return {
    type: ON_BACK,
  };
}
export function clearFoodProfile() {
  return {
    type: ON_BACK,
  };
}
