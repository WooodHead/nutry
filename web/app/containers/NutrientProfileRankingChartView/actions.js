/*
 *
 * NutrientProfileRankingChartView actions
 *
 */

import {
  GET_NUTRIENT_RANKING,
  GET_NUTRIENT_RANKING_SUCCESS,
  GET_NUTRIENT_RANKING_FAILURE,
} from './constants';

export function loadRankings() {
  return {
    type: GET_NUTRIENT_RANKING,
  };
}
export function loadRankingsSuccess(rankingResults) {
  return {
    type: GET_NUTRIENT_RANKING_SUCCESS,
    rankingResults,
  };
}
export function loadRankingsFailure(err) {
  return {
    type: GET_NUTRIENT_RANKING_FAILURE,
    err,
  };
}
