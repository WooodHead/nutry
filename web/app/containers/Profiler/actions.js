/*
 *
 * Profiler actions
 *
 */

import {
  CHANGE_PROFILER_SEARCH,
  SEARCH_COMPLETE,
  SEARCH_FAILURE,
  SEARCH_LOADING,
  ADD_PROFILER_ELEMENT,
  DELETE_PROFILER_ELEMENT,
  UPDATE_PROFILER_ELEMENT,
} from './constants';

export function changeSearch() {
  return {
    type: CHANGE_PROFILER_SEARCH,
  };
}
export function searchComplete(results) {
  if (results) {
    return {
      type: SEARCH_COMPLETE,
      results,
    };
  }
  return {
    type: SEARCH_COMPLETE,
    results: [],
  };
}
export function searchLoading() {
  return {
    type: SEARCH_LOADING,
  };
}
export function searchFailure(error) {
  return {
    type: SEARCH_FAILURE,
    error,
  };
}
export function addProfilerElement() {
  return {
    type: ADD_PROFILER_ELEMENT,
  };
}
export function deleteProfilerElement(id) {
  return {
    type: DELETE_PROFILER_ELEMENT,
    id,
  };
}
export function updateProfilerElement(id, key, value) {
  return {
    type: UPDATE_PROFILER_ELEMENT,
    id,
    key,
    value,
  };
}
