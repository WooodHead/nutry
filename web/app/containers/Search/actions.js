/*
 *
 * SearchB actions
 *
 */

import {
  DEFAULT_ACTION,
  SEARCH_TYPE_CHANGED,
  REFRESH_SEARCH,
  CHANGE_SEARCH,
  SEARCH_COMPLETE,
  SEARCH_FAILURE,
  SEARCH_LOADING,
  PROFILE_SELECTED,
} from './constants';

export function defaultAction() {
  return {
    type: DEFAULT_ACTION,
  };
}
export function changeSearchString(searchString) {
  return {
    type: CHANGE_SEARCH,
    searchString,
  };
}
export function searchRefresh() {
  return {
    type: REFRESH_SEARCH,
  };
}
export function searchComplete(results) {
  if (results) {
    return {
      type: SEARCH_COMPLETE,
      results,
    };
  }
  return {
    type: SEARCH_COMPLETE,
    results: [],
  };
}
export function searchLoading() {
  return {
    type: SEARCH_LOADING,
  };
}
export function searchFailure(error) {
  return {
    type: SEARCH_FAILURE,
    error,
  };
}
export function changeSearchType(searchType) {
  return {
    type: SEARCH_TYPE_CHANGED,
    searchType,
  };
}
export function profileSelected(profileInfo) {
  return {
    type: PROFILE_SELECTED,
    profileInfo,
  };
}
