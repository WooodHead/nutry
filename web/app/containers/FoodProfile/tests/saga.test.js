// import { expectSaga } from 'redux-saga-test-plan';
import { delay } from 'redux-saga';
import { call, takeLatest, put } from 'redux-saga/effects';
import { profileBody } from 'fixtures/foodprofile';

import {
  GET_PROFILE,
} from '../constants';
import { loadProfileSuccess, loadProfileFailure } from '../actions';
import fetchProfile, { getProfile } from '../saga';


describe('Container [FoodProfile] - sagas', () => { /* eslint-disable redux-saga/yield-effects */
  describe('getProfile Saga', () => {
    let getProfileGenerator;
    beforeEach(() => {
      getProfileGenerator = getProfile();
      // select serialNumber
      const selectDescriptor = getProfileGenerator.next('2222222').value;
      expect(selectDescriptor).toMatchSnapshot();
      // call axios getFoodProfile
      const callDescriptor = getProfileGenerator.next().value;
      expect(callDescriptor).toMatchSnapshot();
    });
    it('should dispatch loadProfileSuccess action if the token is recieved and stored successfully', () => {
      const response = profileBody;
      const putDescriptor = getProfileGenerator.next(response).value;
      expect(putDescriptor).toEqual(put(loadProfileSuccess(response)));
    });
    it('should dispatch loadProfileSuccess action if the token is recieved and stored successfully', () => {
      const response = profileBody;
      const putDescriptor = getProfileGenerator.next(response).value;
      expect(putDescriptor).toEqual(put(loadProfileSuccess(response)));
      expect(getProfileGenerator.next().done).toEqual(true);
    });
    it('should dispatch loadProfileSuccess after retry attempts made', () => {
      const response = new Error('Fetch Failure');
      const failAttempts = 1;
      // exceeds failure attempts
      for (let i = 0; i < failAttempts; i += 1) {
        const callDescriptor = getProfileGenerator.throw(response).value;
        expect(callDescriptor).toEqual(call(delay, 1000));
        getProfileGenerator.next();
      }
      // this attempt now succeeds
      getProfileGenerator.next();
      const callDescriptor2 = getProfileGenerator.next().value;
      expect(callDescriptor2).toMatchSnapshot();
      const putDescriptor = getProfileGenerator.next(profileBody).value;
      expect(putDescriptor).toEqual(put(loadProfileSuccess(profileBody)));
    });
    it('should dispatch loadProfileFailure action if too many errors are thrown', () => {
      const response = new Error('Fetch Failure');
      const failAttempts = 2;
      // exceeds failure attempts
      for (let i = 0; i < failAttempts; i += 1) {
        const callDescriptor = getProfileGenerator.throw(response).value;
        expect(callDescriptor).toEqual(call(delay, 1000));
        getProfileGenerator.next();
      }
      // last failure will exist re-try loop
      getProfileGenerator.throw(response);
      const putDescriptor = getProfileGenerator.next().value;
      expect(putDescriptor).toEqual(put(loadProfileFailure(response)));
    });
  });
  describe('fetchProfile Saga', () => {
    const fetchProfileSaga = fetchProfile();
    it('works with unit tests', () => {
      const takeLatestDescriptor = fetchProfileSaga.next().value;
      expect(takeLatestDescriptor).toEqual(takeLatest(GET_PROFILE, getProfile));
    });
  });
  // describe('fetchProfile Saga', () => {
  //   const fetchProfileSaga = fetchProfile();
  //   it('works with next redux events triggering', () => {
  //     const takeLatestDescriptor = fetchProfileSaga.next().value;
  //     expect(takeLatestDescriptor).toEqual(takeLatest(GET_PROFILE, getProfile));
  //     const takeDescriptor = fetchProfileSaga.next().value;
  //     expect(takeDescriptor).toEqual(take([LOCATION_CHANGE, GET_PROFILE_SUCCESS]));
  //   });
  // });
  // describe('fetchProfile Saga', () => {
  //   const fetchProfileSaga = fetchProfile();
  //   it('works with unit tests when hitting cancellation event', () => {
  //     const takeLatestDescriptor = fetchProfileSaga.next().value;
  //     expect(takeLatestDescriptor).toEqual(takeLatest(GET_PROFILE, getProfile));
  //     const takeDescriptor = fetchProfileSaga.next({ type: LOCATION_CHANGE, payload: { done: true } }).value;
  //     expect(takeDescriptor).toEqual(take([LOCATION_CHANGE, GET_PROFILE_SUCCESS]));
  //     const doneDescriptor = fetchProfileSaga.next({ type: LOCATION_CHANGE, payload: { done: true } }).value;
  //     expect(doneDescriptor).toEqual(cancel(...takeLatestDescriptor));
  //   });
  // });
  // describe('fetchProfile Saga', () => {
  //   const fetchProfileSaga = fetchProfile();
  //   it('works with unit tests when not done', () => {
  //     const takeLatestDescriptor = fetchProfileSaga.next().value;
  //     expect(takeLatestDescriptor).toEqual(takeLatest(GET_PROFILE, getProfile));
  //     const takeDescriptor = fetchProfileSaga.next().value;
  //     expect(takeDescriptor).toEqual(take([LOCATION_CHANGE, GET_PROFILE_SUCCESS]));
  //     const doneDescriptor = fetchProfileSaga.next().value;
  //     expect(doneDescriptor).toEqual(takeLatest(GET_PROFILE, getProfile));
  //   });
  // });
});
