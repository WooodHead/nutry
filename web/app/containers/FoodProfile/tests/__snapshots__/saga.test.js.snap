// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container [FoodProfile] - sagas getProfile Saga should dispatch loadProfileFailure action if too many errors are thrown 1`] = `
Object {
  "@@redux-saga/IO": true,
  "SELECT": Object {
    "args": Array [],
    "selector": [Function],
  },
}
`;

exports[`Container [FoodProfile] - sagas getProfile Saga should dispatch loadProfileFailure action if too many errors are thrown 2`] = `
Object {
  "@@redux-saga/IO": true,
  "CALL": Object {
    "args": Array [
      undefined,
    ],
    "context": null,
    "fn": [Function],
  },
}
`;

exports[`Container [FoodProfile] - sagas getProfile Saga should dispatch loadProfileSuccess action if the token is recieved and stored successfully 1`] = `
Object {
  "@@redux-saga/IO": true,
  "SELECT": Object {
    "args": Array [],
    "selector": [Function],
  },
}
`;

exports[`Container [FoodProfile] - sagas getProfile Saga should dispatch loadProfileSuccess action if the token is recieved and stored successfully 2`] = `
Object {
  "@@redux-saga/IO": true,
  "CALL": Object {
    "args": Array [
      undefined,
    ],
    "context": null,
    "fn": [Function],
  },
}
`;

exports[`Container [FoodProfile] - sagas getProfile Saga should dispatch loadProfileSuccess action if the token is recieved and stored successfully 3`] = `
Object {
  "@@redux-saga/IO": true,
  "SELECT": Object {
    "args": Array [],
    "selector": [Function],
  },
}
`;

exports[`Container [FoodProfile] - sagas getProfile Saga should dispatch loadProfileSuccess action if the token is recieved and stored successfully 4`] = `
Object {
  "@@redux-saga/IO": true,
  "CALL": Object {
    "args": Array [
      undefined,
    ],
    "context": null,
    "fn": [Function],
  },
}
`;

exports[`Container [FoodProfile] - sagas getProfile Saga should dispatch loadProfileSuccess after retry attempts made 1`] = `
Object {
  "@@redux-saga/IO": true,
  "SELECT": Object {
    "args": Array [],
    "selector": [Function],
  },
}
`;

exports[`Container [FoodProfile] - sagas getProfile Saga should dispatch loadProfileSuccess after retry attempts made 2`] = `
Object {
  "@@redux-saga/IO": true,
  "CALL": Object {
    "args": Array [
      undefined,
    ],
    "context": null,
    "fn": [Function],
  },
}
`;

exports[`Container [FoodProfile] - sagas getProfile Saga should dispatch loadProfileSuccess after retry attempts made 3`] = `
Object {
  "@@redux-saga/IO": true,
  "CALL": Object {
    "args": Array [
      undefined,
    ],
    "context": null,
    "fn": [Function],
  },
}
`;
